lib/panoptes/client/authentication.rb

```diff
@@ -0,0 +1,73 @@
+require 'deprecate'
+
+module Panoptes
+  class Client
+    module Authentication
+      extend Gem::Deprecate
+      def jwt_payload
+        raise NotLoggedIn unless @auth[:token]
+        payload, = JWT.decode @auth[:token], jwt_signing_public_key, algorithm: 'RS512'
+        payload
```

Comment:
`payload` should be memoized to save the decoding every time you want to access data

Reply:
I actually specifically didn't want to memoize this because a token could expire at any time and we don't know what order the attributes will be accessed

Reply:
isn't the `@auth[:token]` memoized as well ?

Reply:
yes, but `@auth[:token]` is the the version of the credentials that has been encrypted with Panoptes' private key. we keep recalculating payload to see if we are still permitted to decrypt the credentials

Reply:
why wouldn't we it's the same token?

Reply:
the validity period expiration doesn't mean you can't decrypt it. it just means the current timestamp is past the listed one.